New presentation from template as url param

ezequielc opened the pull request:
- Added `templateId` param for new presentation url
  - Added `new` to new presentation url, so it looks more usual. New presentation url supports both `#/editor/workspace//` and `#/editor/workspace/new/`
  - Updated node version to fix build

@alex-deaconu Please review. Thanks

alex-deaconu commented on js/services/svc-editor-factory.js:
```diff
@@ -65,6 +65,35 @@ angular.module('risevision.editorApp.services')
         $rootScope.$broadcast('presentationUpdated');
       };
 
+      factory.newPresentationFromTemplate = function(templateId) {
```
@ezequielc this function (`newPresentationFromTemplate`) practically duplicates the same functionality found in two other functions in this factory: `_addTemplate` and `copyPresentation` (https://github.com/Rise-Vision/editor-app/blob/master/js/services/svc-editor-factory.js#L297). 

Basically I think those 2 functions can be adapted to be used for this scenarios as well: update `_addTemplate` to receive a presentationId rather than a product and I think we're good.

Thoughts? Thanks!

ezequielc replied:
@alex-deaconu Agreed. I will merge them. Thanks

alex-deaconu commented on js/app.js:
```diff
@@ -79,16 +79,21 @@ angular.module('risevision.editorApp', [
                 } else if (!$stateParams.copyPresentation) {
-                  editorFactory.openPresentationProperties();
-                  return editorFactory.newPresentation();
+                  var templateId = $location.search().templateId;
```
@ezequielc as discussed, the way I envision for this to work is to have the following paths:
1. `editor/workspace/cc1997d7-b7a1-414e-9683-a728285e921e/`
for editing an existing presentation

2. `editor/workspace/new/`
for a blank/new presentation (need to update https://github.com/Rise-Vision/editor-app/blob/master/partials/presentation-list.html#L15)

3. `editor/workspace/new/copy/` or `editor/workspace/copy/` or `editor/workspace/new/true`
Copy presentation, use presentation copy from the factory

4. `editor/workspace/new/cc1997d7-b7a1-414e-9683-a728285e921e/` or
`editor/workspace/template/cc1997d7-b7a1-414e-9683-a728285e921e/`
the ID is for the template. 
Use the ID to retrieve the template and copy it. See other comment below.
Also, remove pre-loading code for templates from the Store from https://github.com/Rise-Vision/editor-app/blob/master/js/services/svc-editor-factory.js#L334.

Thoughts? Thanks!

ezequielc replied:
@alex-deaconu I agree with points 1 and 2, but I would use `editor/workspace/new/` with parameters to copy presentation and copy from template. 
I think it would be clearer to have `editor/workspace/new/?copyFromTemplate=cc1997d7-b7a1-414e-9683-a728285e921e`, because we are making the same action (creating a new presentation) but pre-filling it with data from the template. Thoughts? Thanks

alex-deaconu replied:
@ezequielc okay, sounds good. I'm just not a fan of using a variable in `$location.search()` when we can use a route. It seems inconsistent when using UI router. Thanks.

ezequielc commented:
@alex-deaconu I implemented the changes suggested in the first comment (removing `newPresentationFromTemplate`).

As it was taking longer than I planned, I didn't change the routes to match all the cases we listed. I added support for `editor/workspace/new/` and `editor/workspace/new/?copyOf=37135b38-25ba-463e-b885-99d447f44cf9`, so we can deliver the fix. 

We can work on routes for templates later, while working on the card "Templates Shared with You".
Would that be ok?

Please review. Thanks!

alex-deaconu commented:
@ezequielc okay, the only issue I see with that approach is that we would need to update the Store URL twice. 

Otherwise, PR looks good :+1: .

Thanks.
